### DISCUSSION TITLE
Need Guidance for Training Custom YOLO Model with Better Accuracy

### DISCUSSION THREAD

**Discussion by tedo001** [ultralytics/ultralytics#24292] [General]
Hello Ultralytics Team and Community,

I’m currently working on a custom object detection project using YOLO and wanted some guidance regarding improving model accuracy.

Current Setup:
Dataset: Custom labeled images
Model: YOLOv8 / YOLO11
Training Epochs: 100
Image Size: 640
Hardware: GPU / Local PC

Questions:
What are the best augmentation settings for improving real-world detection?
How can I improve small object detection performance?
Is transfer learning recommended for niche datasets?
Any tips for reducing false positives?
Goal:
I want to deploy this model into a real-world application with high speed and good precision.

**Comment by UltralyticsAssistant** [comment]
👋 Hello @tedo001, thank you for your interest in Ultralytics 🚀! This is an automated response to help get things moving quickly, and an Ultralytics engineer will also assist you soon.
We recommend a visit to the [Docs](https://docs.ultralytics.com/) for new users where you can find many [Python](https://docs.ultralytics.com/usage/python/) and [CLI](https://docs.ultralytics.com/usage/cli/) usage examples and where many of the most common questions may already be answered.
Since this is a custom training ❓ question about improving accuracy for a YOLO11 model, please provide as much information as possible, including dataset image examples, class distribution, validation results, confusion matrix, sample predictions, and training logs, and verify you are following our [Tips for Best Training Results](https://docs.ultralytics.com/guides/model-training-tips/).
If you run into a specific 🐛 issue while training or validation, please provide a [minimum reproducible example](https://docs.ultralytics.com/help/minimum-reproducible-example/) to help us debug it.
Join the Ultralytics community where it suits you best. For real-time chat, head to [Discord](https://discord.com/invite/ultralytics) 🎧. Prefer in-depth discussions? Check out [Discourse](https://community.ultralytics.com/). Or dive into threads on our [Subreddit](https://www.reddit.com/r/Ultralytics/) to share knowledge with the community.
## Upgrade
Upgrade to the latest `ultralytics` package including all [requirements](https://github.com/ultralytics/ultralytics/blob/main/pyproject.toml) in a [**Python>=3.8**](https://www.python.org/) environment with [**PyTorch>=1.8**](https://pytorch.org/get-started/locally/) to verify your issue is not already resolved in the latest version:
```bash
pip install -U ultralytics
```
## Environments
YOLO may be run in any of the following up-to-date verified environments (with all dependencies including [CUDA](https://developer.nvidia.com/cuda)/[CUDNN](https://developer.nvidia.com/cudnn), [Python](https://www.python.org/) and [PyTorch](https://pytorch.org/) preinstalled):
- **Notebooks** with free GPU: <a href="https://console.paperspace.com/github/ultralytics/ultralytics"><img src="https://assets.paperspace.io/img/gradient-badge.svg" alt="Run on Gradient"/></a> <a href="https://colab.research.google.com/github/ultralytics/ultralytics/blob/main/examples/tutorial.ipynb"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"></a> <a href="https://www.kaggle.com/models/ultralytics/yolo26"><img src="https://kaggle.com/static/images/open-in-kaggle.svg" alt="Open In Kaggle"></a>
- **Google Cloud** Deep Learning VM. See [GCP Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/google_cloud_quickstart_tutorial/)
- **Amazon** Deep Learning AMI. See [AWS Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/aws_quickstart_tutorial/)
- **Docker Image**. See [Docker Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/docker_image_quickstart_tutorial/) <a href="https://hub.docker.com/r/ultralytics/ultralytics"><img src="https://img.shields.io/docker/pulls/ultralytics/ultralytics?logo=docker" alt="Docker Pulls"></a>
## Status
<a href="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule"><img src="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml/badge.svg" alt="Ultralytics CI"></a>
If this badge is green, all [Ultralytics CI](https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule) tests are currently passing. CI tests verify correct operation of all YOLO [Modes](https://docs.ultralytics.com/modes/) and [Tasks](https://docs.ultralytics.com/tasks/) on macOS, Windows, and Ubuntu every 24 hours and on every commit.

**Reply by tedo001** [reply]
Thank you for the response !

Here are my training details:

* Model: YOLO11s
* Dataset: 17k images
* Classes: 4
* Epochs: 60
* Img size: 640
* Batch size: 16
* GPU:3050rtx

Current Results:
mAP50: 0.91
Precision: 0.88
Recall: 0.84

Main issue:
The model struggles with [small objects detection issue  / false positives / low accuracy model ].
i trained so many time still accuracy is missed 
I’m attaching validation results and sample predictions. I’d appreciate recommendations to improve performance.

**Reply by glenn-jocher** [reply]
Thanks for the details — with `mAP50=0.91`, I’d focus less on more epochs and more on small-object recall: first verify tiny-box annotations are very consistent, then try a pretrained `yolo26s.pt`, increase `imgsz` or use image tiling if VRAM allows, keep augmentation light so small targets are not cropped away, and tune `conf` plus IoU/NMS thresholds to cut false positives; the [small-object guide](https://www.ultralytics.com/blog/how-to-improve-model-map-on-small-objects-a-quick-guide) and [model evaluation insights](https://docs.ultralytics.com/guides/model-evaluation-insights/) are the most relevant next steps, and if you share a few failed predictions plus per-class AP we can suggest the highest-impact change.

**Reply by tedo001** [reply]
ok I will try sir ! and then I come here with a result

**Reply by glenn-jocher** [reply]
Sounds good — when you’ve tested, please share the updated `results.png`, `confusion_matrix.png`, per-class AP, and 2–3 failed small-object predictions so we can pinpoint whether `imgsz`, tiling, or `conf`/IoU tuning is the next best change.

**Reply by tedo001** [reply]
sir ! I used your yolo26x and now accuracy is so good at detection!!

**Reply by glenn-jocher** [reply]
Good to hear `yolo26x` improved detection — before deployment, validate on a fully unseen test split and tune `conf` on real-world images to balance missed detections vs false positives, and if you want to optimize speed next we can help using the [Benchmark mode guide](https://docs.ultralytics.com/modes/benchmark).

**Reply by tedo001** [reply]
I think currently I am so addicted to ultralytics 😅.

**Reply by glenn-jocher** [reply]
Glad it’s working well now — credit to the YOLO community and Ultralytics team; if you want, the next practical step is to benchmark `yolo26x` against smaller variants on your target hardware so you can lock in the best speed/accuracy tradeoff before deployment.

**Reply by tedo001** [reply]
okay sir!!

**Comment by vincere-mori** [comment]
A few concrete things that actually move the needle:

**Small objects:** bump `imgsz` to 1280 if you have the VRAM, that's usually the single biggest win. Also enable `copy_paste` augmentation and run the auto-anchor check to verify your anchor sizes cover your object scale.

**Augmentations for real-world deployment:** keep mosaic on, but lower `degrees` and `shear` if your objects have a natural orientation in the deployment images. Aggressive geometric augmentations hurt when the test distribution is mostly upright/normal angles.

**Transfer learning:** yes, always start from COCO pretrained. If you have less than roughly 500 images per class, freeze the backbone for the first 20 epochs using `freeze=10` in your training args. It prevents the pretrained features from getting destroyed early on.

**False positives:** first check what they actually look like visually. If they're structurally similar to your positives, you need hard negatives in the training set. Run inference on your val set, pull the false positives, label them correctly (or as background), and fold them back in. Adjusting the confidence threshold at inference time helps short-term but does not fix the root cause.

**Reply by glenn-jocher** [reply]
Good suggestions above overall, with one important update: current Ultralytics YOLO models are anchor-free, so I’d skip anchor tuning and instead start from a pretrained `yolo26s.pt` or `yolo26m.pt`, follow the two-stage workflow in the [fine-tuning guide](https://docs.ultralytics.com/guides/finetuning-guide) (`freeze=10` for a short first stage, then unfreeze with lower `lr0`), and for small objects prioritize higher `imgsz` or tiling plus hard-negative images to reduce false positives as outlined in the [small-object accuracy guide](https://www.ultralytics.com/blog/how-to-improve-model-map-on-small-objects-a-quick-guide); if you share your class count and sample image size, we can suggest a tighter starting config.
